source/common/http/header_map_impl.h

```diff
@@ -322,7 +322,8 @@ class HeaderMapImpl : NonCopyable {
   HeaderEntryImpl& maybeCreateInline(HeaderEntryImpl** entry, const LowerCaseString& key);
   HeaderEntryImpl& maybeCreateInline(HeaderEntryImpl** entry, const LowerCaseString& key,
                                      HeaderString&& value);
-  HeaderMap::NonConstGetResult getExisting(const LowerCaseString& key);
+  HeaderMap::NonConstGetResult getExisting(const absl::string_view key);
```

Comment:
Here we change the function's contract by adding a new implicit requirement for a caller that `key` must be lower case. Can we guarantee that or we don't care about casing now?

Reply:
`getExisting(const absl::string_view key)` is a protected method that be used by `get(const LowerCaseString& key)` and `getTracingContext(const absl::string_view key)`. 

The former is the interface of HeaderMap and can always keep the case constraint. The latter is the interface of TracingContext, which is designed to not care about the case of the key.
When developers use TracingContext based on RequestHeaderMapImpl, they should ensure that the key meets the requirements of RequestHeaderMap (key must be lowercase).

Reply:
The design here is currently not perfect. I am thinking about whether it is possible to move Http::LowCaseString to common, and then TracingContext directly uses LowCaseString as the key type.

Reply:
In the design of TracingContext, I hope it is completely protocol-independent, so a simple string_view is used as the key type.
But in actual use, the combination of TracingContext and HeaderMap is not perfect.
In getTracingContext, I must assume that the content of key is a lowercase string. In setTracingContext, I must create a LowCaseString copy of the key.
I am thinking about whether I can directly move LowCaseString to common, and then TracingContext directly uses LowCaseString as the key type.
But I am still struggling with it. Maybe the community can give some opinions.

Reply:
I don't have much experience with non-HTTP tracers and don't know if it's practical to convert to LowerCaseString always :( Will defer to the code owners to decide.